PsfLauncher/PsfPowershellScriptRunner.h

```diff
@@ -194,9 +195,26 @@ class PsfPowershellScriptRunner
 		else
 		{
 			//We don't want to stop on an error and we want to run async
-			std::thread pwrShellThread = std::thread(StartProcess, nullptr, script.commandString.data(), script.currentDirectory.c_str(), script.showWindowAction, script.timeout, m_AttributeList.get());
+			std::thread pwrShellThread = std::thread(StartProcess, nullptr, script.commandString.data(), script.currentDirectory.c_str(), script.showWindowAction, script.timeout, m_AttributeList.get(), &exitCode);
 			pwrShellThread.detach();
 		}
+
+		if (exitCode != ERROR_SUCCESS)
+		{
+			// when powershell fails to run, reset first run of the powershell(when "runOnce" is set) till powershell successfully runs once
+			THROW_IF_FAILED(resetFirstRunStatus(script.shouldRunOnce));
+		}
+
+		DWORD execPolicyFailExitCode = 0x01;
+		if (exitCode == execPolicyFailExitCode)
+		{
+			MessageBoxEx(NULL, (script.scriptPath + std::wstring(L" failed due to an execution policy restriction. To change the executing policy, execute \"Set-ExecutionPolicy -ExecutionPolicy RemoteSigned -Scope LocalMachine\" and re-run the application.")).c_str(), L"Package Support Framework", MB_OK | MB_ICONWARNING, 0);

```

Comment:
how about adding resetFirstRunStatus in execPolicy case

Reply:
Even in case of script errors, intention is to re-run the script on every application launch till it runs successfully